Add file naming conventions documentation to style guide (#292)
IMAP-Science-Operations-Center/imap_processing: docs/source/development/style-guide/naming-conventions.rst

Naming Conventions
------------------

Data Product File Naming Conventions
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The following describes the filename conventions for IMAP data products. These conventions follow the `recommendations
provided by the SPDF <https://spdf.gsfc.nasa.gov/guidelines/filenaming_recommendations.html>`_.

The general filename convention is as follows::

    imap_<instrument>_<datalevel>_<descriptor>_<startdate>_<enddate>_<version>.<format>

* ``<instrument>`` is the IMAP instrument associated with the data product. Acceptable values include: ``codice``,
  ``glows``, ``hi-45``, ``hi-90``, ``hit``, ``idex``, ``lo``, ``mag``, ``swapi``, ``swe``, ``ultra-45``, and
  ``ultra-90``

* ``<datalevel>`` is the data level for the data product.  Acceptable values depend on the instrument:

  * ``codice``: ``l0``, ``l1a``, ``l1b``, ``l2``, ``l3``
  * ``glows``: TBD
  * ``hi-45``: TBD
  * ``hi-90``: TBD
  * ``hit``: TBD
  * ``idex``: TBD
  * ``lo``: TBD
  * ``mag``: TBD
  * ``swapi``: TBD
  * ``swe``: TBD
  * ``ultra-45``: TBD

>   * ``ultra-90``: TBD

Author: Should we explicitly list the supported data levels here?  If so, I will need to dig through our documentation add these. (If anyone knows what they are for the instrument(s) they are working on and could comment here, that would be helpful!)

Reviewer: Probably will be nice. I suggest letting everyone else fill these in during follow-up PRs and not holding it up on you here.

Reviewer 2: MAG: L1A, L1B, L1C, L2pre, L2
GLOWS: L1A, L1B, L2, L3a, L3b, L3c, L3d

That being said, I don't think they need to be included, because I don't think we're checking that in the code...?

Reviewer 3: Yeah, Agree. We don't check in the code yet. But I think having that level of detail is helpful in the document though. If you like to start filling in this PR, this page will give you data level for each instrument:
https://lasp.colorado.edu/galaxy/display/IMAP/Instrument+Science+Data+Processing+Pipelines+Diagrams

Author: Thanks @maxinelasp and @tech3371!  

I agree that having these listed out could provide a nice reference, if nothing else.

> * ``<descriptor>`` stores information specific to the instrument and can store any information that is relevant. For
>   example, it could store the orbit number, pointing number, or data product name.

Reviewer: Add an example here? Do we want orbit number first or last?

```suggestion
* ``<descriptor>`` stores information specific to the instrument and can store any information that is relevant. For
  example, it will typically contain the data product name, and additionally may contain the pointing number.
      ``data-product-123`` for IMAP's 123rd repointing.
```

>   example, it could store the orbit number, pointing number, or data product name.

Reviewer 2: ```suggestion
  example, it could store the orbit number, pointing number, or data product name. This field can have any text or numbers as long as it doesn't include underscores.
```

* ``<startdate>`` is the date of the 'earliest' data within the data product, in the format of ``YYYYMMDD``.
* ``<enddate>`` is the date of the 'latest' data within the data product if the data product spans more than one day.
  If the data does not span multiple days, then the ``<enddate>`` will be the same as the ``<startdate>``. The format
  is also ``YYYYMMDD``.
* ``<version>`` stores the version of the data product in the format is ``vXX-YY``.  See the versioning conventions
  described in the :ref:`data product versioning <data-product-versioning>` documentation for further details.
* ``<format>`` is the data format. For ``l0`` data products, the format is ``pkts``. For data levels ``l1`` and higher,
  the format is ``cdf``.

Here are a few examples of acceptable filenames:

* ``imap_codice_l1a_lo_20261206_20261207_v01-01.cdf``

> * ``imap_swe_l2_descriptor_20261206_20261206_v01-02.cdf``

Author: I thought it might be helpful to add a few examples here, though I will need to determine what `<descriptors>` instruments are using.

Reviewer: Yeah, I agree these examples are nice. Maybe even add text about what they explicitly are? Here are some attempts at that. Feel free to take/leave/modify as you see fit.
```suggestion
A version "01-01" CoDICE L1a data product called "lo" containing data from 2026-12-06 to 2026-12-07
* ``imap_codice_l1a_lo_20261206_20261207_v01-01.cdf``
A SWE L2 data product called "descriptor" containing a single day's worth of data on 2026-12-06. The version "01-02" indicates the software/algorithm version is "01" and the data dependency information has been updated once to make the data version "02".
* ``imap_swe_l2_descriptor_20261206_20261206_v01-02.cdf``
```

Reviewer 2: example MAG file:

imap_mag_l1c_burst_20261206_20261206_v01-01.cdf
